cmd/injector/standalone_injector.go

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy ofthe License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specificlanguage governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package injector
+
+import (
+	"net"
+	"time"
+
+	"github.com/google/gopacket/layers"
+	"github.com/skydive-project/skydive/common"
+	pi "github.com/skydive-project/skydive/packetinjector"
+)
+
+func getNameFromMac(mac string) string {
+	interfaces, _ := net.Interfaces()
+	for _, ifi := range interfaces {
+		if ifi.HardwareAddr.String() == mac {
+			return ifi.Name
+		}
+	}
+	return ""
+}
+
+func injectPacket(pip *pi.PacketInjectionParams) {
+	var rawSocket *common.RawSocket
+	ifName := getNameFromMac(pip.SrcMAC.String())
+	protocol := 3
+
+	if ifName == "" {
+		return
+	}
+
+	rawSocket, err := common.NewRawSocket(ifName, protocol)
+	if err != nil {
+		return
+	}
+
+	newInjector := &pi.PacketInjector{
+		RawSocket: rawSocket,
+		IfName:    ifName,
+	}
+
+	fpg := &pi.ForgedPacketGenerator{
+		PacketInjectionParams: pip,
+		LayerType:             layers.LayerTypeEthernet,
+	}
+	chnl := &pi.Channels{Pipes: make(map[string](chan bool))}
+	pi.DoInjections(pip, chnl, newInjector, fpg, "")
```

Comment:
we should add a mechanism to wait for the end of the injection + adding log to show the progress